File: jediterm-core-mpp/src/jvmMain/java/com/jediterm/core/Platform.java
package com.jediterm.core;

public enum Platform {
    Windows,
    macOS,
    Linux,
    Unknown;

    private static Platform current = detectPlatform();

    private static Platform detectPlatform() {
        String osName = System.getProperty("os.name").toLowerCase();
        if (osName.contains("win")) {
            return Windows;
        } else if (osName.contains("mac")) {
            return macOS;
        } else if (osName.contains("nux") || osName.contains("nix")) {
            return Linux;
        } else {
            return Unknown;
        }
    }

    public static Platform current() {
        return current;
    }

    public static boolean isWindows() {
        return current() == Windows;
    }

    public static boolean isMacOS() {
        return current() == macOS;
    }

    public static boolean isLinux() {
        return current() == Linux;
    }
}

File: jediterm-core-mpp/src/jvmMain/kotlin/com/jediterm/core/Platform.kt
package com.jediterm.core

import java.util.*

enum class Platform {
    Windows,
    macOS,
    Linux,
    Unknown;

    companion object {
        private val current: Platform = detectPlatform()

        private fun detectPlatform(): Platform {
            val osName = System.getProperty("os.name").lowercase(Locale.getDefault())
            if (osName.contains("win")) {
                return Platform.Windows
            } else if (osName.contains("mac")) {
                return Platform.macOS
            } else if (osName.contains("nux") || osName.contains("nix")) {
                return Platform.Linux
            } else {
                return Platform.Unknown
            }
        }

        @JvmStatic
        fun current(): Platform {
            return current
        }

        val isWindows: Boolean
            get() = current() == Platform.Windows

        @JvmStatic
        val isMacOS: Boolean
            get() = current() == Platform.macOS

        val isLinux: Boolean
            get() = current() == Platform.Linux
    }
}
